File: frontend/src/api/dto/gpxPreviewListResponse.ts
import type { GpxPreviewSummaryDto } from './gpxPreviewSummaryDto.ts'

/**
 * Paginated GPX preview list response
 */
export interface GpxPreviewListResponse {
  /** List of GPX previews */
  previews: GpxPreviewSummaryDto[]
  /** Total number of previews */
  total: number
  /** Current page number */
  page: number
  /** Page size */
  size: number
}

File: backend/src/main/java/fr/pedalons/dto/gpx/response/GpxPreviewListResponse.java
package fr.pedalons.dto.gpx.response;

import fr.pedalons.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

/** Paginated list of the current user's analysed GPX files. */
@Schema(description = "Paginated GPX preview list response")
@ValidateSchema
public record GpxPreviewListResponse(
    @Schema(description = "List of GPX previews", required = true)
        List<GpxPreviewSummaryDto> previews,
    @Schema(description = "Total number of previews", required = true) long total,
    @Schema(description = "Current page number", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}
